spec/src/main/resources/META-INF/smithy/bsp/bsp.smithy

```diff
@@ -0,0 +1,1380 @@
+/// but to not exit (otherwise the response might not be delivered correctly to the client). There is a separate exit notification that
+/// asks the server to exit.
+@jsonRequest("build/shutdown")
+operation ShutdownBuild {
+
+}
+
+@jsonNotification("build/exit")
+/// Like the language server protocol, a notification to ask the server to exit its process. The server should exit with success code 0
+/// if the shutdown request has been received before; otherwise with error code 1.
+operation OnBuildShutdown {
+}
+
+
+/// The show message notification is sent from a server to a client to ask the client to display a particular message in the user interface.
+///
+/// A build/showMessage notification is similar to LSP's window/showMessage, except for a few additions like id and originId.
+@jsonNotification("build/showMessage")
+operation OnBuildShowMessage {
+    input: OnBuildShowMessageParams
+}
+
+/// The log message notification is sent from a server to a client to ask the client to log a particular message in its console.
+///
+/// A build/logMessage notification is similar to LSP's window/logMessage, except for a few additions like id and originId.
+@jsonNotification("build/logMessage")
+operation OnBuildLogMessage {
+    input: LogMessageParams
+}
+
+/// The Diagnostics notification are sent from the server to the client to signal results of validation runs.
+///
+/// Diagnostic is defined as it is in the LSP
+///
+/// When reset is true, the client must clean all previous diagnostics associated with the same textDocument and
+/// buildTarget and set instead the diagnostics in the request. This is the same behaviour as PublishDiagnosticsParams
+/// in the LSP. When reset is false, the diagnostics are added to the last active diagnostics, allowing build tools to
+/// stream diagnostics to the client.
+///
+/// It is the server's responsibility to manage the lifetime of the diagnostics by using the appropriate value in the reset field.
+/// Clients generate new diagnostics by calling any BSP endpoint that triggers a buildTarget/compile, such as buildTarget/compile, buildTarget/test and buildTarget/run.
+@jsonNotification("build/publishDiagnostics")
+operation OnBuildPublishDiagnostics {
+    input: PublishDiagnosticsParams
+}
+
+/// The workspace build targets request is sent from the client to the server to ask
+/// for the list of all available build targets in the workspace.
+@jsonRequest("workspace/buildTargets")
+operation WorkspaceBuildTargets {
+    input: WorkspaceBuildTargetsParams
+    output: WorkspaceBuildTargetsResult
+}
+
+/// The `reload` request is sent from the client to instruct the build server to reload
+/// the build configuration. This request should be supported by build tools that keep
+/// their state in memory. If the `reload` request returns with an error, it's expected
+/// that other requests respond with the previously known "good" state.
+@jsonRequest("workspace/reload")
+operation WorkspaceReload {
```

Comment:
Another question about `errors` in `operation`s. So in here I know it says:

> code and message in case an error happens during reload. For example, when the build configuration is invalid.

So are these supposed to actually be encoded in here or no? Like we know that this can error and that it's supposed to have a code and a message. I supposed if we'd do that we'd need to create a `GenericBspError` structure or something like that _if we wanted to do it_?

Reply:
> I supposed if we'd do that we'd need to create a GenericBspError structure or something like that if we wanted to do it? 

The way it'd look like would probably be : 

``` 
@jsonRPCError(-32130) // or whatever 
structure ReloadError {
    message: String 
    whateverElse: Something 
}
```

and the data contained by this ReloadError` shape would be filling the `data` of the error payload.

Reply:
Gotcha, makes sense.